lib/AuthStateManager.ts

```diff
@@ -131,7 +140,7 @@ class AuthStateManager {
             isPending = shouldEvaluateIsPending();
           }
           let promise = isAuthenticated 
-            ? isAuthenticated(this._sdk)
+            ? isAuthenticated(this._sdk, { isPending, accessToken, idToken })
```

Comment:
Thinking about this some more, I think what I'd actually like is a function `transformAuthState` or `mutateAuthState` that allows me to modify the auth state before it is emitted. This would allow me to modify `isAuthenticated` or any other values on the state, as well as add other values to the state that my application might want.  I would prefer this instead of the `isAuthenticated` option, as it is much more powerful and flexible.

Reply:
I am not a fan of let devs mutate the internal authState, as they may not know well of how the internal state is evaluated. It can be complex to explain which states are new or not. 
I still prefer to the current approach to provided devs a restricted flow, then devs add callback that returns value for the authState. I think we can give the passed in options a better name, like `evaluatedStates` to clarify what's in the passed object.